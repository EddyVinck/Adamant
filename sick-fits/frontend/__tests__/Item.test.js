import ItemComponent from "../components/Item";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

const fakeItem = {
  id: "ABC123",
  title: "A test item",
  price: 5000,
  description: "This is a really nice item.",
  image: "sunglasses.jpg",
  largeImage: "largesunglasses.jpg"
};

describe("<Item/>", () => {
  it("renders pricetag and title properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const PriceTag = wrapper.find("PriceTag");
    expect(PriceTag.children().text()).toBe("â‚¬50");
    // console.log(wrapper.debug());

    expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
  });

  it("renders the image properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const img = wrapper.find("img");
    const { src, alt } = img.props();
    expect(src).toBe(fakeItem.image);
    expect(alt).toBe(fakeItem.title);
    // console.log(img.debug());
  });

  it("renders out the buttons properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = wrapper.find(".buttonList");

    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find("Link").exists()).toBe(true);
    expect(buttonList.find("AddToCart").exists()).toBe(true);
    expect(buttonList.find("DeleteItem").exists()).toBe(true);
  });

  it("renders and matches the snapshot", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
