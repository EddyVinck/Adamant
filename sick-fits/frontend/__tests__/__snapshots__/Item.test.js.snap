// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A test item"
    src="sunglasses.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A test item
      </a>
    </Link>
  </Title>
  <PriceTag>
    â‚¬50
  </PriceTag>
  <p>
    This is a really nice item.
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ðŸ–‹
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
